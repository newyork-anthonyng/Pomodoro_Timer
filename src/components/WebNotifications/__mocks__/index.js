import { PureComponent } from 'react';

class WebNotifications extends PureComponent {
  constructor() {
    super();

    this.createNotification = jest.fn();
  }

  render() {
    return null;
  }
}

export default WebNotifications;
