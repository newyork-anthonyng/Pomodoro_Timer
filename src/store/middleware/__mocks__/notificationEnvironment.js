window.Notification = jest.fn();
window.Notification.permission = "granted";