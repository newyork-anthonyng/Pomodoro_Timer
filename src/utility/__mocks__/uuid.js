const uuid = (function() {
  return function() {
    return 42;
  };
})();

export default uuid;
