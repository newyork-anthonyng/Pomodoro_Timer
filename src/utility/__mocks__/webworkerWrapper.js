const webworkerWrapper = jest.fn(() => {
  return {
    postMessage: jest.fn(),
  };
});

export default webworkerWrapper;
