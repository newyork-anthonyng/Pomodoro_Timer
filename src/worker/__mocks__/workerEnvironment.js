global._events_ = {};

global.addEventListener = jest.fn((name, callback) => {
    global._events_[name] = callback;
});

global.postMessage = jest.fn();

beforeEach(() => {
    global._events = {};

    global.addEventListener.mockClear();
    global.postMessage.mockClear();
});