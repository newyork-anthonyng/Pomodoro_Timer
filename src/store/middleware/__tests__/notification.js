import "../__mocks__/notificationEnvironment";
import notificationMiddleware from "../notification";

describe("notificationMiddleware", () => {
    let store;
    let action;
    let state;

    beforeEach(() => {
        state = {
            isPlaying: true,
            time: 42
        };
        store = {
            getState: () => state
        };
        action = { type: "FOO_BAR" };
    });

    it("should call the next function with the action", () => {
        const next = jest.fn();

        notificationMiddleware(store)(next)(action);

        expect(next).toHaveBeenCalledTimes(1);
        expect(next.mock.calls[0][0]).toEqual(action);
    });

    it("should create notification if timer stopped and time is expired", () => {
        const next = () => {
            state.isPlaying = false;
            state.time = 0;
        };

        expect(window.Notification).toHaveBeenCalledTimes(0);

        notificationMiddleware(store)(next)(action);

        expect(window.Notification).toHaveBeenCalledTimes(1);
        expect(window.Notification.mock.calls[0]).toMatchSnapshot();
    });
});