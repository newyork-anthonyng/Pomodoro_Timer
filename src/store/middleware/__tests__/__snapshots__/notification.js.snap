// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notificationMiddleware should create notification if timer stopped and time is expired 1`] = `
Array [
  "Pomodoro Timer expired",
  Object {
    "body": "",
    "requireInteraction": true,
  },
]
`;
